### PR TITLE
Topics returned from find_topic should be distinct Entities

### PR DESCRIPTION
**Pull request by mitza-oci**

### CODE REVIEW

**Review by jrw972 — APPROVED** [review]

File: dds/DCPS/DomainParticipantImpl.cpp [hunk 1]
```diff
@@ -623,24 +623,31 @@ DomainParticipantImpl::delete_topic_i(
...
+          break;
+        }
+      }
+      if (entry == 0) {
         if (DCPS_debug_level > 0) {
           ACE_ERROR((LM_ERROR,
-                     ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::delete_topic_i, ")
-                     ACE_TEXT("%p\n"),
-                     ACE_TEXT("find")));
+                     ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::delete_topic_i, not found\n")));
```
Comment (jwillemsen) [inline dds/DCPS/DomainParticipantImpl.cpp:637]: Any logging of id/name helpful?

**Review by jwillemsen — APPROVED** [review]

File: dds/DCPS/DomainParticipantImpl.cpp (hunk 1 above)
Comment (simpsont-oci) [reply dds/DCPS/DomainParticipantImpl.cpp:637]:
```suggestion
                     ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::delete_topic_i() - topic '%s' not found\n"), topic_name.in()));
```

Comment (jwillemsen) [reply dds/DCPS/DomainParticipantImpl.cpp:637]: topic_name is char*, so %C

Comment (simpsont-oci) [reply dds/DCPS/DomainParticipantImpl.cpp:637]: Ah, sorry, I saw CORBA so I assumed ACE_TCHAR.

**Pull request merged**